### PR TITLE
8299635: Hotspot update for deprecated sprintf in Xcode 14

### PR DESCRIPTION
**Pull request by gdams**
Changes to the patch:

The following files don't exist in jdk17u
- `test/hotspot/jtreg/runtime/Thread/libAsyncExceptionOnMonitorEnter.cpp` (https://github.com/openjdk/jdk/commit/4e20a03786094f1eb0c662fe8f281295bf1bac83 JDK19+)
- `test/hotspot/jtreg/serviceability/jvmti/events/ThreadEnd/threadend01/libthreadend01.cpp` (https://github.com/openjdk/jdk/commit/9583e3657e43cc1c6f2101a64534564db2a9bd84 JDK19+)
- `test/hotspot/jtreg/serviceability/jvmti/events/ThreadStart/threadstart01/libthreadstart01.cpp` (https://github.com/openjdk/jdk/commit/9583e3657e43cc1c6f2101a64534564db2a9bd84 JDK19+)

The changes didn't apply to `src/hotspot/share/opto/node.cpp` as https://github.com/openjdk/jdk/commit/33ed0365c3ed182a9d063e1701fe69bfb72dfa2e is JDK20+

The changes didn't apply to `src/hotspot/share/utilities/globalDefinitions.hpp` as https://github.com/openjdk/jdk/commit/04f02ac6b2ce496b86642987bb7e25d21b52a5b6 is JDK19+

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8299635](https://bugs.openjdk.org/browse/JDK-8299635) needs maintainer approval

### Issue
 * [JDK-8299635](https://bugs.openjdk.org/browse/JDK-8299635): Hotspot update for deprecated sprintf in Xcode 14 (**Enhancement** - P4 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2663/head:pull/2663` \
`$ git checkout pull/2663`

Update a local copy of the PR: \
`$ git checkout pull/2663` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2663/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2663`

View PR using the GUI difftool: \
`$ git pr show -t 2663`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2663.diff">https://git.openjdk.org/jdk17u-dev/pull/2663.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2663#issuecomment-2210440980)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+s
```
Comment (gdams) [inline test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp:2]: this feels like an accidental `s` added to the comment but it's the same in openjdk/jdk https://github.com/openjdk/jdk/blob/4ec1ae109710aa150e27acf5706475d335c4655c/test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp#L1-L5 so needs fixing there

**Comment by gdams** [comment]
### Adoptium Build CI
  
  - https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-x64-temurin/535/
  - https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-aarch64-temurin/474/

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp (hunk 1 above)
Comment (jerboaa) [reply test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp:2]: Thanks. Created https://github.com/openjdk/jdk/pull/20051

**Review by jerboaa — APPROVED** [review]
Looks OK to me. File differences were as you said. That was helpful info, thanks!

**Comment by gdams** [comment]
/approval request This change applies cleanly other than some missing files in JDK17u. All tests pass. This is also crucial to allow us to upgrade the GH actions workflow to move away from the deprecated macos-11 VMs.

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp (hunk 1 above)
Comment (gnu-andrew) [reply test/hotspot/jtreg/vmTestbase/nsk/jvmti/unit/functions/rawmonitor/rawmonitor.cpp:2]: Good catch and thanks @jerboaa for fixing in trunk. I'll open a backport to 21u.

**Pull request closed without merging**